core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -695,6 +695,11 @@ sealed abstract private[data] class NonEmptyListInstances0 extends NonEmptyListI
     new NonEmptyListPartialOrder[A] {
       val A0 = A
     }
+
+  implicit val catsDataNonEmptyReducibleNonEmptyList: NonEmptyReducible[NonEmptyList, List] =
+    new NonEmptyReducible[NonEmptyList, List]() {
+      override def split[A](fa: NonEmptyList[A]): (A, List[A]) = (fa.head, fa.tail)
```

Comment:
now that we have this we could write split in [here](https://github.com/typelevel/cats/blob/22c3f79175902b529d4718b3d4b915e1e6cd1423/core/src/main/scala/cats/data/NonEmptyList.scala#L542) in terms of it. Something like
```scala
override def split[A](fa: NonEmptyList[A]): (A, List[A]) = catsDataNonEmptyReducibleNonEmptyList.split(fa)
```
and we could do the same for NonEmptyVector. The reason I'm suggesting this is I wouldn't want for someone to change the implementation of split in one place and get some inconsistent behaviour on the other instances.

Reply:
Thanks for the great point. Code duplication is never good.